### ISSUE TITLE
Nested configuration is not resolved in cloud formation resolver

### ISSUE THREAD

**Issue by smacker** [serverless/serverless#10294]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

I'm trying to upgrade to the new serverless from v1 and I'm facing an issue with cloud formation resolver.

Keeping the config as it produces incorrect output with a warning:
```
Cannot resolve variable at "provider.environment.SECOND_LEVEL_NESTED": Approached variable syntax error in resolved value "${cf.ap-southeast-1:oms-ext-dev.SnsOrderTopicArn}": Missing variable closing bracket in ${cf.ap-southeast-1:
```
(see "Command output" section below for the full output)

Changing the syntax to use `cf(region):` as described in the current documentation instead of the old `cf.region:` produces the following error:
```
 Serverless Error ----------------------------------------

  Invalid variable reference syntax for variable cf(ap-southeast-1):oms-ext-dev.SnsOrderTopicArn. You can only reference env vars, options, & files. You can check our docs for more info.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              16.9.1
     Framework Version:         2.67.0 (local)
     Plugin Version:            5.5.1
     SDK Version:               4.3.0
     Components Version:        3.18.1

Serverless: Deprecation warning: Variables resolver reports following resolution errors:
              - Cannot resolve variable at "provider.environment.SECOND_LEVEL_NESTED": 1 validation error detected: Value '${file(${opt:stage}' at 'stackName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z][-a-zA-Z0-9]*|arn:[-a-zA-Z0-9:/._+]*
            From a next major this will be communicated with a thrown error.
            Set "variablesResolutionMode: 20210326" in your service config, to adapt to new behavior now
            More Info: https://www.serverless.com/framework/docs/deprecations/#NEW_VARIABLES_RESOLVER
```

### Service configuration (serverless.yml) content

```yaml
# serverless.yml
service: aws-node-http-api-project

provider:
  name: aws
  environment:
    SECOND_LEVEL_NESTED: ${cf.${opt:region}:${file(variables.yml):EXTENSION_STACK_NAME}.SnsOrderTopicArn}

# variables.yml
EXTENSION_STACK_NAME: ${file(${opt:stage}.yml):EXTENSION_STACK_NAME}

# dev.yml
EXTENSION_STACK_NAME: oms-ext-dev
```

### Command name and used flags

yarn sls print --stage dev --region ap-southeast-1

### Command output

```shell
service:
  name: aws-node-http-api-project
provider:
  stage: dev
  region: ap-southeast-1
  name: aws
  environment:
    SECOND_LEVEL_NESTED: arn:aws:sns:ap-southeast-1:907551155229:dev-oms-order
  versionFunctions: true

Serverless: Deprecation warning: Variables resolver reports following resolution errors:
              - Cannot resolve variable at "provider.environment.SECOND_LEVEL_NESTED": Approached variable syntax error in resolved value "${cf.ap-southeast-1:oms-ext-dev.SnsOrderTopicArn}": Missing variable closing bracket in ${cf.ap-southeast-1:
            From a next major this will be communicated with a thrown error.
            Set "variablesResolutionMode: 20210326" in your service config, to adapt to new behavior now
            More Info: https://www.serverless.com/framework/docs/deprecations/#NEW_VARIABLES_RESOLVER
```

### Environment information

```shell
Framework Core: 2.67.0 (local)
Plugin: 5.5.1
SDK: 4.3.0
Components: 3.18.1
```

**Comment by medikoo** [comment]
@smacker great thanks for reporting! Indeed there's an issue in resolving variables that have _param_ or _address_ part behind variables which resolve with values that are further constructed with variables.

Fix landed in #10296. It'll be fixed with next release

**Comment by smacker** [comment]
That was fast! Thank you!
